core/templates/dev/head/components/forms/schema-based-editors/schema-based-unicode-editor.directive.ts

```diff
@@ -133,8 +86,56 @@ angular.module('oppia').directive('schemaBasedUnicodeEditor', [
             return $sce.trustAsHtml(
               $filter('convertUnicodeWithParamsToHtml')(ctrl.localValue));
           };
-        }
-      ]
+          ctrl.$onInit = function() {
+            if (ctrl.uiConfig() && ctrl.uiConfig().coding_mode) {
+              // Flag that is flipped each time the codemirror view is
+              // shown. (The codemirror instance needs to be refreshed
+              // every time it is unhidden.)
+              ctrl.codemirrorStatus = false;
+              var CODING_MODE_NONE = 'none';
+
+              ctrl.codemirrorOptions = {
+                // Convert tabs to spaces.
+                extraKeys: {
+                  Tab: function(cm) {
+                    var spaces = Array(
+                      cm.getOption('indentUnit') + 1).join(' ');
+                    cm.replaceSelection(spaces);
+                    // Move the cursor to the end of the selection.
+                    var endSelectionPos = cm.getDoc().getCursor('head');
+                    cm.getDoc().setCursor(endSelectionPos);
+                  }
+                },
+                indentWithTabs: false,
+                lineNumbers: true
+              };
+
+              if (ctrl.isDisabled()) {
+                ctrl.codemirrorOptions.readOnly = 'nocursor';
+              }
+              // Note that only 'coffeescript', 'javascript', 'lua', 'python',
+              // 'ruby' and 'scheme' have CodeMirror-supported syntax
+              // highlighting. For other languages, syntax highlighting will not
+              // happen.
+              if (ctrl.uiConfig().coding_mode !== CODING_MODE_NONE) {
+                ctrl.codemirrorOptions.mode = ctrl.uiConfig().coding_mode;
+              }
+
+              setTimeout(function() {
+                ctrl.codemirrorStatus = !ctrl.codemirrorStatus;
+              }, 200);
+
+              // When the form view is opened, flip the status flag. The
```

Comment:
Should this be outside $onInit?

Also maybe replace the setTimeout calls with $timeout.

Reply:
This is inside the initial if block in $onInit, so it cannot be moved outside.

Updated $timeout to setTimeout.

Reply:
Read the comment incorrectly. Fixed this - changed setTimeout to $timeout.